### PR TITLE
typing: add type hints in a few files (Bug 1759890)

### PR DESCRIPTION
**Pull request by cgsheeh**
Also removed a few functions that are unused from `commit_message.py`.

### CODE REVIEW

File: landoapi/commit_message.py [hunk 1]
```diff
@@ -185,35 +188,28 @@ def replace_first_reviewer(matchobj):
         return commit_summary.strip() + "\n" + commit_description
 
 
-def strip_commit_metadata(s):
+def strip_commit_metadata(message: str) -> str:
```
Comment (zzzeid) [inline landoapi/commit_message.py:191]: Is this used anywhere?

File: landoapi/commit_message.py [hunk 2]
```diff
@@ -185,35 +188,28 @@ def replace_first_reviewer(matchobj):
...
-        joiner = "\n"
-    else:
-        raise TypeError("do not know type of commit message: %s" % type(s))
-
-    return joiner.join(lines)
+    return "\n".join(lines)
 
 
-def parse_commit_id(s):
+def parse_commit_id(message: str) -> Optional[str]:
```
Comment (zzzeid) [inline landoapi/commit_message.py:207]: Is this used anywhere?

File: landoapi/ui.py [hunk 3]
```diff
@@ -12,7 +14,7 @@
 class LandoUISubsystem(Subsystem):
     name = "lando_ui"
 
-    def ready(self):
+    def ready(self) -> Union[bool, str]:
```
Comment (zzzeid) [inline landoapi/ui.py:17]: Could probably use the newer `bool | str` if you import from \_\_future\_\_.

**Review by zzzeid — CHANGES_REQUESTED** [review]

File: landoapi/commit_message.py (hunk 2 above)
Comment (cgsheeh) [reply landoapi/commit_message.py:207]: Hm, neither of the functions you called out are used... I wonder if that's something we could configure Flake8 to look for?

In the meantime I'll remove them.

**Review by zzzeid — APPROVED** [review]
lgtm. Final commit message may have to be edited to reflect removed functions.

**Pull request merged**